### PR TITLE
CRUD for fieldOfActivity

### PR DESCRIPTION
**Pull request by EdiOancea**

### CODE REVIEW

File: app/controllers/crudController.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const errorService = require('../services/error');
+
+module.exports = name => {
```
Comment (lgnstn) [inline app/controllers/crudController.js]: I think a better idea is to add a `constructor` to the `Controller` class and pass the service name in the constructor. Then you can export the class here and you instantiate it in the actual controller file.

File: app/repositories/address.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class AddressRepository {
+  getByAddress(address) {
+    return db.addresses.findOne({
+      where: { ...address },
+      paranoid: false,
+    });
+  }
+
+  get(id) {
```
Comment (lgnstn) [inline app/repositories/address.js]: You can do the thing you did for the controllers here, for the repositories as well. You can have a base repository class where you implement the common methods and these specific repositories can extend the base one.

File: app/controllers/address.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const crudController = require('./crudController');
+
+module.exports = crudController('address');
```
Comment (lgnstn) [inline app/controllers/address.js]: I would create a class `AddressController` which extends the base controller, then `module.exports = new AddressController('address');`.

File: app/services/fieldOfActivity.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+const fieldOfActivityRepository = require('../repositories/fieldOfActivity');
+const error = require('../services/error');
+
+class FieldOfActivityService {
```
Comment (lgnstn) [inline app/services/fieldOfActivity.js]: Same thing for services :)

**Review by lgnstn — CHANGES_REQUESTED** [review]

**Comment by EdiOancea** [comment]
@lgnstn updated

**Review by lgnstn — APPROVED** [review]

**Pull request merged**